src/main/scala/org/broadinstitute/dsde/rawls/model/JsonSupport.scala

```diff
@@ -1,57 +1,145 @@
+sealed trait AttributeListSerializer {
+  def writeListType(obj: Attribute): JsValue
+  def readListType(json: JsValue): Attribute
+
+  def writeAttribute(obj: Attribute): JsValue
+  def readAttribute(json: JsValue): Attribute
+}
+
+//Serializes attribute lists to e.g. [1,2,3]
+//When reading in JSON, assumes that [] is an empty value list, not an empty ref list
+trait PlainArrayAttributeListSerializer extends AttributeListSerializer {
+  override def writeListType(obj: Attribute): JsValue = obj match {
+    //lists
+    case AttributeValueEmptyList => JsArray()
+    case AttributeValueList(l) => JsArray(l.map(writeAttribute):_*)
+    case AttributeEntityReferenceEmptyList => JsArray()
+    case AttributeEntityReferenceList(l) => JsArray(l.map(writeAttribute):_*)
+  }
+
+  override def readListType(json: JsValue): Attribute = json match {
+    case JsArray(a) =>
+      val attrList: Seq[Attribute] = a.map(readAttribute)
+      attrList match {
+        case e: Seq[_] if e.isEmpty => AttributeValueEmptyList
+        case v: Seq[AttributeValue @unchecked] if attrList.map(_.isInstanceOf[AttributeValue]).reduce(_&&_) => AttributeValueList(v)
+        case r: Seq[AttributeEntityReference @unchecked] if attrList.map(_.isInstanceOf[AttributeEntityReference]).reduce(_&&_) => AttributeEntityReferenceList(r)
+        case _ => throw new DeserializationException("illegal array type")
+      }
+    case _ => throw new DeserializationException("unexpected json type")
+  }
+}
+
+//Serializes attribute lists to e.g. { "itemsType" : "AttributeValue", "items" : [1,2,3] }
+trait TypedAttributeListSerializer extends AttributeListSerializer {
+  val LIST_ITEMS_TYPE_KEY = "itemsType"
+  val LIST_ITEMS_KEY = "items"
+  val LIST_OBJECT_KEYS = Set(LIST_ITEMS_TYPE_KEY, LIST_ITEMS_KEY)
+
+  val VALUE_LIST_TYPE = "AttributeValue"
+  val REF_LIST_TYPE = "EntityReference"
+
+  def writeListType(obj: Attribute): JsValue = obj match {
+    //lists
+    case AttributeValueEmptyList => writeAttributeList(VALUE_LIST_TYPE, Seq.empty[AttributeValue])
+    case AttributeValueList(l) => writeAttributeList(VALUE_LIST_TYPE, l)
+    case AttributeEntityReferenceEmptyList => writeAttributeList(REF_LIST_TYPE, Seq.empty[AttributeEntityReference])
+    case AttributeEntityReferenceList(l) => writeAttributeList(REF_LIST_TYPE, l)
+  }
+
+  def readListType(json: JsValue): Attribute = json match {
+    case JsObject(members) if LIST_OBJECT_KEYS subsetOf members.keySet => readAttributeList(members)
+
+    case _ => throw new DeserializationException("unexpected json type")
+  }
+
+  def writeAttributeList[T <: Attribute](listType: String, list: Seq[T]): JsValue = {
+    JsObject( Map(LIST_ITEMS_TYPE_KEY -> JsString(listType), LIST_ITEMS_KEY -> JsArray(list.map( writeAttribute ).toSeq:_*)) )
+  }
+
+  def readAttributeList(jsMap: Map[String, JsValue]) = {
+    val attrList: Seq[Attribute] = jsMap(LIST_ITEMS_KEY) match {
```

Comment:
it's only called from `readListType()` which has already checked the key does exist